### ISSUE TITLE
Snapshot 3391630 fails to build on DragonFly: liblibc - binary operation + cannot be applied

### ISSUE THREAD

**Issue by mneumann** [rust-lang/rust#30596]
When I compile I get the following error:

```
src/liblibc/src/unix/bsd/freebsdlike/dragonfly.rs:3:32: 3:36 error: binary operation `+` cannot be applied to type `_` [E0369]
src/liblibc/src/unix/bsd/freebsdlike/dragonfly.rs:3 pub const SIGSTKSZ: ::size_t = 8192 /* MINSIGSTKSZ */ + 32768;
                                                                                   ^~~~
src/liblibc/src/macros.rs:26:9: 26:75 note: in this expansion of __cfg_if_apply! (defined in src/liblibc/src/macros.rs)
```

I can fix it by performing the addition myself and expanding it to: 40960

**Comment by huonw** [comment]
@dhuseby is the brokenness #30628 refers to?

**Comment by dhuseby** [comment]
@mneumann if you try building at the most recent snapshot revision 3391630, you'll see different errors.

**Comment by dhuseby** [comment]
@mneumann I'm getting a bunch of errors like this:

```
rustc: x86_64-unknown-dragonfly/stage2/lib/rustlib/x86_64-unknown-dragonfly/lib/libsyntax
../src/libsyntax/util/interner.rs:95:34: 95:44 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/interner.rs:95 #[derive(Clone, PartialEq, Hash, PartialOrd)]
                                                                      ^~~~~~~~~~
../src/libsyntax/util/interner.rs:95:34: 95:44 note: in this expansion of #[derive_PartialOrd] (defined in ../src/libsyntax/util/interner.rs)
../src/libsyntax/util/interner.rs:95:28: 95:32 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/interner.rs:95 #[derive(Clone, PartialEq, Hash, PartialOrd)]
                                                                ^~~~
../src/libsyntax/util/interner.rs:95:28: 95:32 note: in this expansion of #[derive_Hash] (defined in ../src/libsyntax/util/interner.rs)
../src/libsyntax/util/interner.rs:95:17: 95:26 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/interner.rs:95 #[derive(Clone, PartialEq, Hash, PartialOrd)]
                                                     ^~~~~~~~~
../src/libsyntax/util/interner.rs:95:17: 95:26 note: in this expansion of #[derive_PartialEq] (defined in ../src/libsyntax/util/interner.rs)
../src/libsyntax/util/interner.rs:95:10: 95:15 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/interner.rs:95 #[derive(Clone, PartialEq, Hash, PartialOrd)]
                                              ^~~~~
../src/libsyntax/util/interner.rs:95:10: 95:15 note: in this expansion of #[derive_Clone] (defined in ../src/libsyntax/util/interner.rs)
../src/libsyntax/util/parser.rs:16:28: 16:30 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:16 #[derive(Debug, PartialEq, Eq)]
                                                              ^~
../src/libsyntax/util/parser.rs:16:28: 16:30 note: in this expansion of #[derive_Eq] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/util/parser.rs:16:17: 16:26 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:16 #[derive(Debug, PartialEq, Eq)]
                                                   ^~~~~~~~~
../src/libsyntax/util/parser.rs:16:17: 16:26 note: in this expansion of #[derive_PartialEq] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/util/parser.rs:16:10: 16:15 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:16 #[derive(Debug, PartialEq, Eq)]
                                            ^~~~~
../src/libsyntax/util/parser.rs:16:10: 16:15 note: in this expansion of #[derive_Debug] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/util/parser.rs:66:28: 66:30 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:66 #[derive(Debug, PartialEq, Eq)]
                                                              ^~
../src/libsyntax/util/parser.rs:66:28: 66:30 note: in this expansion of #[derive_Eq] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/util/parser.rs:66:17: 66:26 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:66 #[derive(Debug, PartialEq, Eq)]
                                                   ^~~~~~~~~
../src/libsyntax/util/parser.rs:66:17: 66:26 note: in this expansion of #[derive_PartialEq] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/util/parser.rs:66:10: 66:15 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/util/parser.rs:66 #[derive(Debug, PartialEq, Eq)] 
                                            ^~~~~
../src/libsyntax/util/parser.rs:66:10: 66:15 note: in this expansion of #[derive_Debug] (defined in ../src/libsyntax/util/parser.rs)
../src/libsyntax/diagnostics/registry.rs:13:10: 13:15 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/diagnostics/registry.rs:13 #[derive(Clone)]
                                                     ^~~~~
../src/libsyntax/diagnostics/registry.rs:13:10: 13:15 note: in this expansion of #[derive_Clone] (defined in ../src/libsyntax/diagnostics/registry.rs)
../src/libsyntax/diagnostics/metadata.rs:31:37: 31:51 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/diagnostics/metadata.rs:31 #[derive(PartialEq, RustcDecodable, RustcEncodable)]
                                                                                ^~~~~~~~~~~~~~
../src/libsyntax/diagnostics/metadata.rs:31:37: 31:51 note: in this expansion of #[derive_RustcEncodable] (defined in ../src/libsyntax/diagnostics/metadata.rs)
../src/libsyntax/diagnostics/metadata.rs:31:21: 31:35 error: unknown `allow` attribute: `unused_qualifications`, #[deny(unknown_lints)] on by default
../src/libsyntax/diagnostics/metadata.rs:31 #[derive(PartialEq, RustcDecodable, RustcEncodable)]
                                                                ^~~~~~~~~~~~~~
```

**Comment by dhuseby** [comment]
AFAICT, for some reason the allow attribute is not defined/enabled in this build.  I'm trying to figure out where the attribute code is defined in the compiler so I can figure out why #[allow] is unknown.

**Comment by dhuseby** [comment]
Oh! it's the unused_qualifications allow attribute that is missing.  Hrm, what is this error happening only on DragonflyBSD at this revision?

**Comment by mneumann** [comment]
@dhuseby : IIRC I built 3391630, and for me it worked. With which version are you building the snapshot? Maybe we use different version of rust?

**Comment by dhuseby** [comment]
@mneumann it should be downloading the last snapshot 1af31d4 to build it with.

**Comment by dhuseby** [comment]
(I don't have rust installed on the host)

**Comment by dhuseby** [comment]
@mneumann if you can successfully build a snapshot at 3391630, then clone my https://github.com/dhuseby/rust-manual-snapshots repo, add it there, send me a PR and then send a PR to rust and I'll merge it.  Just ping @alexcrichton with the link to the file in the rust-manual-snapshots repo and ask him to upload it.

**Comment by mneumann** [comment]
@dhuseby : I posted a link to my snapshot here https://github.com/rust-lang/rust/pull/30643 (you can read in that thread the problem I encountered). http://www.ntecs.de/downloads/rust/rust-stage0-2015-12-18-3391630-dragonfly-x86_64-e74d79488e88ac2de3bd03afd5959d2ae6e2b628.tar.bz2. I will clone your repo and push it there.

**Comment by dhuseby** [comment]
@mneumann i'll do it, it's morning for me and I can merge it all.

**Comment by dhuseby** [comment]
@mneumann i saw your PR, did you make a PR for rust to update src/snapshot.txt ?

**Comment by dhuseby** [comment]
@mneumann it looks like you haven't done a rust PR yet.  i'll do that.

**Comment by dhuseby** [comment]
@mneumann PR #30781

**Comment by dhuseby** [comment]
@mneumann what version of dragonfly bsd are you running?  i recently updated my build machine to the latest 4.4 version.  maybe that's why you can build it and i can't?

**Comment by mneumann** [comment]
@dhuseby I am running HEAD, but at the time I was compiling, it should have been pretty close to 4.4.

**Comment by dhuseby** [comment]
fixed by #30781
